restcomm/restcomm.dao/src/main/java/org/mobicents/servlet/restcomm/dao/mybatis/MybatisGeolocationDao.java

```diff
@@ -0,0 +1,213 @@
+/*
+ * TeleStax, Open Source Cloud Communications
+ * Copyright 2011-2016, Telestax Inc and individual contributors
+ * by the @authors tag.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.mobicents.servlet.restcomm.dao.mybatis;
+
+import static org.mobicents.servlet.restcomm.dao.DaoUtils.readDateTime;
+import static org.mobicents.servlet.restcomm.dao.DaoUtils.readInteger;
+import static org.mobicents.servlet.restcomm.dao.DaoUtils.readSid;
+import static org.mobicents.servlet.restcomm.dao.DaoUtils.readString;
+import static org.mobicents.servlet.restcomm.dao.DaoUtils.readUri;
+import static org.mobicents.servlet.restcomm.dao.DaoUtils.writeDateTime;
+import static org.mobicents.servlet.restcomm.dao.DaoUtils.writeSid;
+import static org.mobicents.servlet.restcomm.dao.DaoUtils.writeUri;
+import static org.mobicents.servlet.restcomm.dao.DaoUtils.readGeolocationType;
+import static org.mobicents.servlet.restcomm.dao.DaoUtils.readLong;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.ibatis.session.SqlSession;
+import org.apache.ibatis.session.SqlSessionFactory;
+import org.joda.time.DateTime;
+import org.mobicents.servlet.restcomm.dao.GeolocationDao;
+import org.mobicents.servlet.restcomm.entities.Geolocation;
+import org.mobicents.servlet.restcomm.entities.Sid;
+
+/**
+ * @author fernando.mendioroz@telestax.com (Fernando Mendioroz)
+ *
+ */
+public class MybatisGeolocationDao implements GeolocationDao {
+
+    private static final String namespace = "org.mobicents.servlet.sip.restcomm.dao.GeolocationDao.";
+    private final SqlSessionFactory sessions;
+
+    public MybatisGeolocationDao(final SqlSessionFactory sessions) {
+        super();
+        this.sessions = sessions;
+    }
+
+    @Override
+    public void addGeolocation(Geolocation gl) {
+        final SqlSession session = sessions.openSession();
+        try {
+            session.insert(namespace + "addGeolocation", toMap(gl));
+            session.commit();
```

Comment:
Noted @jaimecasero. Again, if we agree this approach could bring trouble, then we should study all MybatisXDao classes, as this one follows the same design of all of them (adapted to its own needs of course).